File: src/main/scala/frc.robot/Robot.scala
package frc.robot

import edu.wpi.first.wpilibj.TimedRobot
import frc.robot.ArmSystem.Simulation._
import frc.robot.ArmSystem._
import frc.robot.PidUpdater._


/**
 * This is a sample program to demonstrate the use of elevator simulation with existing code.
 */
class Robot extends TimedRobot {
  override def robotInit() {
    ArmSystem.Simulation
    PidUpdater.values
  }

  override def simulationPeriodic() {
    if(isOperatorControlEnabled) runSimulation()
  }


  override def teleopPeriodic() {
    updatePid(values)
    applyPidOutput(activeEntry, setpointEntry)
  }

  // This just makes sure that our simulation code knows that the motor's off.
  override def disabledInit(): Unit = neutral
}

File: src/main/kotlin/frc.robot/Robot.kt
package frc.robot

import edu.wpi.first.wpilibj.TimedRobot

class Robot : TimedRobot() {
    override fun robotInit() {
        PidUpdater
        ArmSystem
    }

    override fun simulationPeriodic() = ArmSystem.simulationPeriodic()

    override fun teleopPeriodic() {
        PidUpdater.apply {
            ArmSystem.applyOutput(active, setpoint)
            ArmSystem.updatePID(kp, ki, kd)
        }

    }

    override fun disabledInit() {
        // This just makes sure that our simulation code knows that the motor's off.
        ArmSystem.neutral()
    }

}